lib/ansible/vars/fact_cache.py

```diff
@@ -21,7 +21,7 @@ class FactCache(MutableMapping):
 
     def __init__(self, *args, **kwargs):
 
-        self._plugin = cache_loader.get(C.CACHE_PLUGIN)
+        self._plugin = cache_loader.get(C.CACHE_PLUGIN, class_only=True)()
```

Comment:
I'm not sure how this was intended to be used and therefore what the right fix is. The alternative solution for this (and [this](https://github.com/ansible/ansible/pull/50446/files#diff-a8385538c61f1318541f0276f3e1ec7bR278)) is https://github.com/ansible/ansible/pull/40105/files#diff-cb51550dcfdbd27a3f224be2de15e2e8R396. Either the config definition loading must happen first to work with config or class_only needs to be used so the definitions are loaded first.